packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,197 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:ui';
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/services.dart';
 
-/// A leaf node in the focus tree that can receive focus.
+import 'binding.dart';
+import 'focus_traversal.dart';
+import 'framework.dart';
+
+/// Signature of a callback used by [Focus.onKey] and [FocusScope.onKey]
+/// to receive key events.
+///
+/// The [node] is the node that received the event.
+typedef FocusOnKeyCallback = bool Function(FocusNode node, RawKeyEvent event);
+
+/// An object that can be used by a stateful widget to obtain the keyboard focus
+/// and to handle keyboard events.
+///
+/// {@template flutter.widgets.focus_manager.focus.advancedWarning}
+/// **If you are not implementing a widget which handles focused input directly
+/// (e.g. implementing your own text field or other widget primitive), or
+/// keeping [FocusNode] references so that you can control the focus from
+/// another part of the application, you typically will not need to manage a
+/// [FocusNode] directly, as it will be handled by the [Focus] and [FocusScope]
+/// widgets or by the widget primitives themselves. Please see those widgets for
+/// typical usage.**
+/// {@endtemplate}
 ///
-/// The focus tree keeps track of which widget is the user's current focus. The
-/// focused widget often listens for keyboard events.
+/// [FocusNode]s are persistent objects which form a "focus tree" that is a
```

Comment:
the quote marks around "focus tree" make it sound like it's not a focus tree? In which case, what is it, and can we use that term instead?

But I think it _is_ a focus tree. Maybe use `_focus tree_` instead of `"focus tree"`?

Reply:
ah, yeah, ok. to establish a term use italics rather than quotes.